src/Engine/ProtoCore/Utils/NodeUtils.cs

```diff
@@ -24,6 +26,9 @@ public static ProtoCore.AST.AssociativeAST.AssociativeNode Clone(ProtoCore.AST.A
         {
             if (rhsNode is ProtoCore.AST.AssociativeAST.IdentifierNode)
             {
+                if(rhsNode is TypedIdentifierNode)
```

Comment:
would it make sense to move it out of 'if(rhsNode is TypedIdentifierNode)'
and just have:
if (rhsNode is ProtoCore.AST.AssociativeAST.IdentifierNode){}
elseif(rhsNode is TypedIdentifierNode) {}